Firmware/Marlin_main.cpp

```diff
@@ -7601,16 +7603,50 @@ SERIAL_PROTOCOLPGM("\n\n");
 
     /*!
     ### M601 - Pause print <a href="https://reprap.org/wiki/G-code#M601:_Pause_print">M601: Pause print</a>
+    Without any parameters it will park the extruder to default.
+    #### Usage
+
+         M601 [ X | Y | Z | S ]
+
+    #### Parameters
+     - `X` - X position to park at (otherwise X_PAUSE_POS 50) these are saved until change or reset.
+     - `Y` - Y position to park at (otherwise Y_PAUSE_POS 190) these are saved until change or reset.
+     - `Z` - Z raise before park (otherwise Z_PAUSE_LIFT 20) these are saved until change or reset.
+     - `S` - Set values [S0 = set default | S1 = set values only without pausing]
     */
     /*!
-    ### M125 - Pause print (TODO: not implemented)
+
+    ### M125 - Pause print <a href="https://reprap.org/wiki/G-code#M125:_Pause_print">M125: Pause print</a>
+    Without any parameters it will park the extruderthe extruder to default.
+    #### Usage
+
+         M125 [ X | Y | Z | S ]
+
+    #### Parameters
+     - `X` - X position to park at (otherwise X_PAUSE_POS 50) these are saved until change or reset.
+     - `Y` - Y position to park at (otherwise Y_PAUSE_POS 190 these are saved until change or reset.
+     - `Z` - Z raise before park (otherwise Z_PAUSE_LIFT 20) these are saved until change or reset.
+     - `S` - Set values [S0 = set to default values | S1 = sset values only without pausing]
     */
     /*!
     ### M25 - Pause SD print <a href="https://reprap.org/wiki/G-code#M25:_Pause_SD_print">M25: Pause SD print</a>
     */
     case 25:
+    case 125:
     case 601:
     {
+        if (code_seen('X')) pause_position[1] = code_value();
+        if (code_seen('Y')) pause_position[2] = code_value();
+        if (code_seen('Z')) pause_position[3] = code_value();
+        if (code_seen('S')) {
+            if ( code_value_uint8() == 0 ) {
+                pause_position[1] = X_PAUSE_POS;
+                pause_position[2] = Y_PAUSE_POS;
+                pause_position[3] = Z_PAUSE_LIFT;
```

Comment:
`pause_position[3]` is accessing memory outside the array.

Code probably should be:

```c
pause_position[X_AXIS] = X_PAUSE_POS;
pause_position[Y_AXIS] = Y_PAUSE_POS;
pause_position[Z_AXIS] = Z_PAUSE_LIFT;
```